feat: Automatic Summary feature DOCS-695 (#2110)
codacy/docs: docs/repositories-configure/integrations/github-integration.md

![Automatic Summary on GitHub](images/github-integration-automatic-summary.png)

!!! note
    -   This feature is only available for GitHub.
    -   This feature uses [Amazon Bedrock](https://aws.amazon.com/bedrock/) within Codacy's existing infrastructure. No information is shared with any other third party or used to train AI models.
    -   Summaries are generated using the pull request title, branch name, commit messages, and changes diff.

To enable this feature, add the following to the [Codacy configuration file](../codacy-configuration-file.md) `codacy.yaml` in the root of your repository:

```yaml
reviews:
  high_level_summary: true
```

You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. This file will be used as the default configuration for all repositories in the organization and overridden by repository-specific configuration files.

Once enabled, summaries will be created when pull requests are opened and updated at each commit to reflect any changes to the pull request.

Pull requests opened by bots, such as Dependabot, are ignored.

> 

Author: ```suggestion
### common problems(?)
seeing duplicated comments with the summary => likely cause is multiple configured webhooks to Codacy. solution: delete all but one webhook
```

![AI-enhanced comment on GitHub](images/github-integration-ai-comment.png)

### Automatic summaries

> Even when not written by developers, Codacy can provide an high-level summary for each pull request.

Reviewer: I'm not a fan of this sentence but I don't have a good suggestion.

Author: ```suggestion
Codacy can give a clear, high-level summary for each pull request, even if it's not written by developers.```

Reviewer: Yeah that's better, I just have a tiny suggestion
```suggestion
Codacy can provide a clear, high-level summary for each pull request, even if one is not written by developers.
```

> Even when not written by developers, Codacy can provide an high-level summary for each pull request.

Reviewer: ```suggestion
Codacy can provide a clear, high-level summary for each pull request, even if one is not written by developers.
```

Codacy generates an overview of the changes in the pull request so that any reviewer can understand it's intent and impact.

![Automatic Summary on GitHub](images/github-integration-automatic-summary.png)

!!! note
    -   This feature is only available for GitHub.
    -   This feature uses AWS Bedrock within Codacy's existing infrastructure. No information is shared with any other third party or used to train AI models.
    -   Summaries are generated using the pull request title, branch name, commit messages, and changes diff.

Enabling this feature requires a [Codacy configuration file](../codacy-configuration-file.md).
Create a file named `codacy.yaml` in the root of your repository with the following content:

```yaml
reviews:
  high_level_summary: true
```

You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. (This file will be used as the default configuration for all repositories in the organization, and overriden by the repository-specific configuration file.)

Once enabled, summaries will be created when pull requests are opened and kept up-to-date to reflect any changes to the pull request.

> Pull requests opened by Bot account (such as Dependabot) are automatically ignored.

Reviewer: ```suggestion
Pull requests opened by Bot account (such as Dependabot) are automatically ignored.
```
```suggestion
Pull requests opened by Bot accounts (such as Dependabot) are automatically ignored.
```

![AI-enhanced comment on GitHub](images/github-integration-ai-comment.png)

## Generating automatic pull request summaries

!!! info "This is a preview feature"
    This is an upcoming Codacy feature. If you're interested, ask us for early access.

Codacy can provide a clear, high-level summary of the code changes introduced by a pull request, based on the committed code.
Codacy generates an overview of the changes in the pull request so that any reviewer can understand its intent and impact.

![Automatic Summary on GitHub](images/github-integration-automatic-summary.png)

!!! note
    -   This feature is only available for GitHub.

>     -   This feature uses [Amazon Bedrock](https://aws.amazon.com/bedrock/) within Codacy's existing infrastructure. No information is shared with any other third party or used to train AI models.

Author: ```suggestion
    -   This feature uses only AWS services within Codacy's existing infrastructure. No information is shared with any other third party or used to train AI models.
```

    -   Summaries are generated using the pull request title, branch name, commit messages, and changes diff.

To enable this feature, add the following to the [Codacy configuration file](../codacy-configuration-file.md) `codacy.yaml` in the root of your repository:

```yaml
reviews:
  high_level_summary: true
```

You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. This file will be used as the default configuration for all repositories in the organization and overridden by repository-specific configuration files.

Once enabled, summaries will be created when pull requests are opened and updated at each commit to reflect any changes to the pull request.

Pull requests opened by bots, such as Dependabot, are ignored.
If you see duplicated comments posted by Codacy on the same pull request, please ensure that your repository only has one configured webhook for Codacy.

> ## See also

Author: ```suggestion

## See also
```

![Automatic Summary on GitHub](images/github-integration-automatic-summary.png)

!!! note
    -   This feature is only available for GitHub.
    -   This feature uses only AWS services within Codacy's existing infrastructure. No information is shared with any other third party or used to train AI models.
    -   Summaries are generated using the pull request title, branch name, commit messages, and changes diff.

To enable this feature, add the following to the [Codacy configuration file](../codacy-configuration-file.md) `codacy.yaml` in the root of your repository:

```yaml
reviews:
  high_level_summary: true
```

You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. This file will be used as the default configuration for all repositories in the organization and overridden by repository-specific configuration files.

Once enabled, summaries will be created when pull requests are opened and updated at each commit to reflect any changes to the pull request.

Pull requests opened by bots, such as Dependabot, are ignored.

> If you see duplicated comments posted by Codacy on the same pull request, please ensure that your repository only has one configured webhook for Codacy.

Author: ```suggestion

If you see duplicated comments posted by Codacy on the same pull request, please ensure that your repository only has one configured webhook for Codacy.
```

![AI-enhanced comment on GitHub](images/github-integration-ai-comment.png)

## Generating automatic pull request summaries

!!! info "This is a preview feature"

>     This is an upcoming Codacy feature. If you're interested, ask us for early access.

Reviewer 2: ```suggestion
    This is an upcoming Codacy feature. If you're interested, contact <a href="mailto:support@codacy.com">support@codacy.com</a> for early access.
```
